src/main/java/appeng/items/misc/WrappedGenericStack.java

```diff
@@ -142,6 +143,10 @@ public boolean overrideOtherStackedOnMe(ItemStack itemInSlot, ItemStack otherSta
                 long amount = unwrapAmount(itemInSlot);
                 long inserted = heldContainer.insert(what, amount, Actionable.MODULATE);
 
+                if (what instanceof AEFluidKey && player.getLevel().isClientSide) {
```

Comment:
```suggestion
                // Check client to avoid duplicate sounds in singleplayer
                if (player.getLevel().isClientSide) {
```

The instanceof check is not necessary anymore, and the comment will help us understand why this check exists if we look at it again in a few months.

Reply:
You're right about the isntanceof not being necessary. I didn't think about that. I also think the comment is a good idea,